packages/core/src/modules/dids/domain/verificationMethod/EcdsaSecp256k1VerificationKey2019.ts

```diff
@@ -43,9 +43,17 @@ export function isEcdsaSecp256k1VerificationKey2019(
  * Get a key from a EcdsaSecp256k1VerificationKey2019 verification method.
  */
 export function getKeyFromEcdsaSecp256k1VerificationKey2019(verificationMethod: EcdsaSecp256k1VerificationKey2019) {
-  if (!verificationMethod.publicKeyBase58) {
-    throw new CredoError('verification method is missing publicKeyBase58')
+  if (verificationMethod.publicKeyBase58) {
+    return Key.fromPublicKeyBase58(verificationMethod.publicKeyBase58, KeyType.K256)
+  }
+  if (verificationMethod.publicKeyMultibase) {
```

Comment:
Don't we have this check on level higher in getKeyFromVerificationMethod? We can parse this generically and thus don't have to support it for each key type separately (you can look at JsonWebKey2020 as an example, as it behaves the same)

Reply:
From what I am seeing the `getKeyFromVerificationMethod` function will check if it is multikey in type and then attempt to process the multikey encoding. The issue I was running into in testing was that this function was not the function used everywhere for resolving keys from verification methods. Specifically while processing JSON-LD elsewhere there is code attempting to resolve a specific key type from a verification method that is of type multikey and so we would fail to resolve there. Aca-py by default will now use the multikey encoding and I believe there is a push to make that the standard encoding. I think having this on per-key basis and in the `getKeyFromVerificationMethod` function is more fault tolerant for resolving keys and is the only way to ensure the key you are resolving is of a specific type.

Reply:
Hmm okay -- ideally we update the calls that use the specific getKeyFromVerificationMethod and use the generic one instead. that way we encourage moving towards this approach rather than adding the multikey to format/key-type specific transformers

Reply:
I added a change that checks if the type is multikey or if the multikey property is defined then to use that on the multikey resolution. After merging main I can no longer find the place that used the ed25519 key resolution directly so hopefully that should now be resolved.